DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -24,54 +35,107 @@ public IActionResult Index()
+                        break;
+                    }
+
+                    adminUser = null;
+                }
+
+                if (adminUser == null && delegateUsers.Any(du => !du.Approved) )
+                {
+                    return View("AccountNotApproved");
+                }
+            }
+
+            var claims = GetClaimsForSignIn(adminUser, delegateUsers.FirstOrDefault(du => du.Approved));
             var claimsIdentity = new ClaimsIdentity(claims, "Identity.Application");
             var authProperties = new AuthenticationProperties
             {
                 AllowRefresh = true,
-                IsPersistent = true,
+                IsPersistent = model.RememberMe,
                 IssuedUtc = DateTime.UtcNow
             };
             HttpContext.SignInAsync("Identity.Application", new ClaimsPrincipal(claimsIdentity), authProperties);
             return RedirectToAction("Index", "Home");
         }
+
+        private List<Claim> GetClaimsForSignIn(AdminUser? adminUser, DelegateUser? delegateUser)
+        {
+            var claims = new List<Claim>
+            {
+                new Claim(ClaimTypes.Email, adminUser?.EmailAddress ?? delegateUser?.EmailAddress),
+                new Claim(CustomClaimTypes.UserCentreId, adminUser?.CentreId.ToString() ?? delegateUser?.CentreId.ToString()),
+                new Claim(CustomClaimTypes.UserCentreManager, adminUser?.IsCentreManager.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserCentreAdmin, adminUser?.IsCentreAdmin.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserUserAdmin, adminUser?.IsUserAdmin.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserContentCreator, adminUser?.IsContentCreator.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserAuthenticatedCm, adminUser?.IsContentManager.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserPublishToAll, adminUser?.PublishToAll.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserCentreReports, adminUser?.SummaryReports.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.LearnCandidateId, delegateUser?.Id.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.LearnUserAuthenticated, (delegateUser != null).ToString()),
+                new Claim(CustomClaimTypes.AdminCategoryId, adminUser?.CategoryId.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.IsSupervisor, adminUser?.IsSupervisor.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.IsTrainer, adminUser?.IsTrainer.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.IsFrameworkDeveloper, adminUser?.IsFrameworkDeveloper.ToString() ?? "False"),
+                new Claim(CustomClaimTypes.UserCentreName, adminUser?.CentreName ?? delegateUser?.CentreName)
+            };
+
+            var firstName = delegateUser?.FirstName ?? adminUser?.FirstName;

```

Comment:
Just wanted to check that these are in the correct order given most of the above are admin ?? delegate

Reply:
The order of these doesn't really matter as far as I am aware, as the old code's order for these will change depending on which account is found first. But I might as well make it consistent with the rest.